### PR TITLE
[Playwright] New tests for kits deactivation

### PR DESCRIPTION
**Pull request by aweng98**
Added three new tests:

1. `tests/dsm/kitUploadFlow/kit-deactivation-from-error.spec.ts`
2. `tests/dsm/kitUploadFlow/kit-deactivation-from-kits-without-label.spec.ts`
3. `tests/dsm/kitUploadFlow/kit-deactivation-from-queue.spec.ts`

Refactored to remove lots of duplicated code by creating new `dsm/pages/kits-page-base.ts` which all kits-* pages should extends from.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -924,7 +924,7 @@ workflows:
           <<: *filter-develop-branch
           env: dev
           test_suite: UNKNOWN
-          parallelism_num: 2
+          parallelism_num: 1
```
Comment (aweng98) [inline .circleci/config.yml:927]: Set to 1, to keep error/failure investigation simple.

File: playwright-e2e/dsm/pages/kits-page-base.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import { KitTypeEnum } from 'dsm/component/kitType/enums/kitType-enum';
+import { waitForNoSpinner, waitForResponse } from 'utils/test-utils';
+import { StudyEnum } from 'dsm/component/navigation/enums/selectStudyNav-enum';
+import { KitType } from 'dsm/component/kitType/kitType';
+import { logInfo } from 'utils/log-utils';
+import { KitsColumnsEnum } from './kitsInfo-pages/enums/kitsColumns-enum';
+import Modal from 'dsm/component/modal';
+import { assertTableHeaders } from 'utils/assertion-helper';
+
+export default abstract class KitsPageBase extends DsmPageBase {
```
Comment (aweng98) [inline playwright-e2e/dsm/pages/kits-page-base.ts:13]: `KitsPageBase.ts` is the new class that helps to remove duplicated code in kits pages.

File: playwright-e2e/dsm/pages/participant-list-page.ts [hunk 3]
```diff
@@ -93,12 +93,12 @@ export default class ParticipantListPage {
...
     const saveModal = new Modal(this.page);
+    await expect(saveModal.toLocator()).toBeVisible();
     expect(await saveModal.getHeader()).toBe('Please enter a name for your filter');
     await saveModal.getInput({ label: 'Filter Name' }).fill(viewName);
-    console.log(`New filter name: ${viewName}`);
     await Promise.all([
-      saveModal.getButton({ label: /Save Filter/ }).click(),
-      waitForResponse(this.page, {uri: '/ui/saveFilter?'})
+      waitForResponse(this.page, {uri: '/ui/saveFilter'}),
+      saveModal.getButton({ label: /Save Filter/ }).click()
```
Comment (aweng98) [inline playwright-e2e/dsm/pages/participant-list-page.ts:101]: The order of promises matters inside the Promise.all.

File: playwright-e2e/playwright.config.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ const testConfig: PlaywrightTestConfig = {
   /* Fail the build on CI if you accidentally left test.only in the source code. */
   forbidOnly: !!CI,
   retries: CI ? 1 : 0,
-  workers: CI ? 2 : 3,
+  workers: CI ? 1 : 2,
```
Comment (aweng98) [inline playwright-e2e/playwright.config.ts:47]: Set to 1, to keep error/failure investigation simple.

File: playwright-e2e/tests/dsm/kitUploadFlow/blood-kit-prefix-check.spec.ts [hunk 5]
```diff
@@ -218,9 +217,8 @@ test.describe.serial('Blood Kit Upload', () => {
     await trackingScanPage.save({ verifySuccess: false });
 
     // Enter the same pair again to trigger the scan error
+    const errMsg = `Error occurred sending this scan pair!  Kit ${kitLabel} was already associated with tracking id ${trackingLabel}`;
     await expect(page.locator('//h3[contains(@class, "Color--warn")]')).toHaveText('Error - Failed to save all changes');
-    await expect(page.locator('//p[contains(@class, "Color--warn")]')).toHaveText(
-      `Error occurred sending this scan pair!  Error occured for Kit Label "${kitLabel}" ` +
-      'For more information please contact your DSM developer');
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/kitUploadFlow/blood-kit-prefix-check.spec.ts:224]: Error msg has been changed by Zim.

**Review by Rinchan20 — APPROVED** [review]

**Pull request merged**